### PR TITLE
[SPARK-46580][TESTS] Regenerate benchmark results

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This aims to regenerate benchmark results on `master` branch to monitor and verify the changes as a part of Apache spark 4.0.0 preparation.

### Why are the changes needed?

Since 2023.9, many changes are merged.
- #43065

Also, Java versions are changed.
```
- OpenJDK 64-Bit Server VM 17.0.8+7-LTS on Linux 5.15.0-1046-azure
+ OpenJDK 64-Bit Server VM 17.0.9+9-LTS on Linux 5.15.0-1053-azure
```
```
- OpenJDK 64-Bit Server VM 21+35 on Linux 5.15.0-1046-azure
+ OpenJDK 64-Bit Server VM 21.0.1+12-LTS on Linux 5.15.0-1053-azure
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @LuciferYang

File: sql/core/benchmarks/BloomFilterBenchmark-jdk21-results.txt [hunk 1]
```diff
@@ -2,191 +2,191 @@
...
-OpenJDK 64-Bit Server VM 21+35 on Linux 5.15.0-1046-azure
-Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
+OpenJDK 64-Bit Server VM 21.0.1+12-LTS on Linux 5.15.0-1053-azure
+AMD EPYC 7763 64-Core Processor
 Read a row from 100M rows:                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Without bloom filter, blocksize: 2097152           1291           1323          45         77.4          12.9       1.0X
-With bloom filter, blocksize: 2097152              1065           1079          20         93.9          10.6       1.2X
+Without bloom filter, blocksize: 2097152            869            901          52        115.1           8.7       1.0X
+With bloom filter, blocksize: 2097152               890            895           7        112.4           8.9       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/BloomFilterBenchmark-jdk21-results.txt:22]: This could be a regression. Let me take a look.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon . Let me merge this because this contains only benchmark result changes.

**Comment by LuciferYang** [comment]
late LGTM

**Comment by dongjoon-hyun** [comment]
Thank you, @LuciferYang !

**Comment by yaooqinn** [comment]
late LGTM

**Pull request closed without merging**